plex/player.py

```diff
@@ -174,10 +176,48 @@ def _startPlayback(self):
             )
             self._duration = toMilliseconds(self.getTotalTime())
 
+            # register settings
+            settings = self._mediaProvider.prepareSettings()
+            if not settings:
+                log('failed to load settings for {} ({}) playing from {}' \
+                    .format(self._item.title, self._file, mediaProvider2str(self._mediaProvider)), xbmc.LOGWARNING)
+                self._reset()
+                return
+
+            # load external subtitles
+            if settings.getBool(SETTINGS_PROVIDER_PLAYBACK_ENABLE_EXTERNAL_SUBTITLES):
+                self._addExternalSubtitles(plexServer.PlexServer())
+
         else:
             self._reset()
 
 
+    def _addExternalSubtitles(self, plexServer):
+        '''Add external subtitles to the player'''
+        if not self._item:
+            return
+
+        external_subtitles = self._item.subtitleStreams()
+        if external_subtitles:
+            for subtitle in external_subtitles:
+                self.addSubtitle(
+                    plexServer.url(subtitle.key, includeToken=True),
+                    str(subtitle.title),
+                    str(subtitle.language),
+                    subtitle.selected
+                )
+                log('external subtitle "{}" [{}] at index {} added for "{}" ({}) from media provider {}' \
+                    .format(
+                        str(subtitle.title),
```

Comment:
I'm not good in Python. Are these explicit calls to `str()` necessary?

Reply:
They are because if they are not filled plexapi returns None. I make them strings so that "None" is passed to kodi in such situations

Reply:
Alternative can be to define a literal (`UNKNOWN`) at the top of the file and just use:

```python
subtitle.title if subtitle.title else UNKNOWN
```

Or pass the items through a function that would do the same. Let me know which alternative you prefer

Reply:
Decided to use the above approach since subtitles will look native by showing `Unknown` instead of `None`.